src/funding/card/config.test.jsx

```diff
@@ -0,0 +1,120 @@
+/* @flow */
+import { describe, test, expect } from "vitest";
+import { FUNDING, COMPONENTS } from "@paypal/sdk-constants/src";
+
+import { getCardConfig } from "./config";
+
+const getEligibility = ({
+  eligible = true,
+  branded = false,
+  vendors = {},
+} = {}) => ({
+  card: {
+    eligible,
+    branded,
+    vendors,
+  },
+});
+
+const getWallet = ({ card = [] } = {}) => ({
+  card: {
+    instruments: card,
+  },
+  credit: {
+    instruments: [],
+  },
+  paypal: {
+    instruments: [],
+  },
+  venmo: {
+    instruments: [],
+  },
+});
+
+describe("card eligibility", () => {
```

Comment:
Can we add a test for if `card-fields` is requested and there is a vaulted card as another safe guard for any potential changes to the ordering?

Reply:
ooooh good call

Reply:
these tests kinda suck, its easy to get them wrong. We need something like Property Testing to really make sure every edge case is covered